freqtrade/configuration/arguments.py

```diff
@@ -110,87 +101,122 @@ def _build_subcommands(self) -> None:
 
         # Add backtesting subcommand
-        backtesting_cmd = subparsers.add_parser('backtesting', help='Backtesting module.')
+        backtesting_cmd = subparsers.add_parser('backtesting', help='Backtesting module.',
+                                                parents=[_common_parser, _strategy_parser])
         backtesting_cmd.set_defaults(func=start_backtesting)
         self._build_args(optionlist=ARGS_BACKTEST, parser=backtesting_cmd)
 
         # Add edge subcommand
-        edge_cmd = subparsers.add_parser('edge', help='Edge module.')
+        edge_cmd = subparsers.add_parser('edge', help='Edge module.',
+                                         parents=[_common_parser, _strategy_parser])
         edge_cmd.set_defaults(func=start_edge)
         self._build_args(optionlist=ARGS_EDGE, parser=edge_cmd)
 
         # Add hyperopt subcommand
-        hyperopt_cmd = subparsers.add_parser('hyperopt', help='Hyperopt module.')
+        hyperopt_cmd = subparsers.add_parser('hyperopt', help='Hyperopt module.',
+                                             parents=[_common_parser, _strategy_parser],
+                                             )
         hyperopt_cmd.set_defaults(func=start_hyperopt)
         self._build_args(optionlist=ARGS_HYPEROPT, parser=hyperopt_cmd)
 
         # add create-userdir subcommand
         create_userdir_cmd = subparsers.add_parser('create-userdir',
-                                                   help="Create user-data directory.")
+                                                   help="Create user-data directory.",
+                                                   )
         create_userdir_cmd.set_defaults(func=start_create_userdir)
         self._build_args(optionlist=ARGS_CREATE_USERDIR, parser=create_userdir_cmd)
 
         # Add list-exchanges subcommand
         list_exchanges_cmd = subparsers.add_parser(
             'list-exchanges',
-            help='Print available exchanges.'
+            help='Print available exchanges.',
+            parents=[_common_parser],
         )
         list_exchanges_cmd.set_defaults(func=start_list_exchanges)
         self._build_args(optionlist=ARGS_LIST_EXCHANGES, parser=list_exchanges_cmd)
 
         # Add list-timeframes subcommand
         list_timeframes_cmd = subparsers.add_parser(
             'list-timeframes',
-            help='Print available ticker intervals (timeframes) for the exchange.'
+            help='Print available ticker intervals (timeframes) for the exchange.',
+            parents=[_common_parser],
         )
         list_timeframes_cmd.set_defaults(func=start_list_timeframes)
         self._build_args(optionlist=ARGS_LIST_TIMEFRAMES, parser=list_timeframes_cmd)
 
         # Add list-markets subcommand
         list_markets_cmd = subparsers.add_parser(
             'list-markets',
-            help='Print markets on exchange.'
+            help='Print markets on exchange.',
+            parents=[_common_parser],
```

Comment:
While i've added this now to fix the broken tests, i'm not sure if including the parent parser makes sense.

I did intentionally not do this for `create-userdir` - so it does not accept `--config` as argument - which means that no config file is not loaded (only the basic one), even if `config.json` does exist, fixing some oddities with these commands.

I'll leave the decision to you if this makes sense for these 2 commands ...

Reply:
It's okay for now, let's leave it as it is, thanks.
👍